server/controllers/documents.js

```diff
@@ -0,0 +1,243 @@
+   * @param {Object} res - Response object
+   * @returns {Object} Response object
+   */
+  listAll(req, res) {
+    let query = {};
+    query.limit = Number(req.query.limit) !== 'NaN'? req.query.limit : 10;
+    query.offset = Number(req.query.limit) !== 'NaN'? req.query.offset : 0;
+    if (Helpers.isAdmin(req)) {
+      db.Document
+        .findAll(query)
+        .then(documents => res.status(200).send(documents));
+    } else {
+      query = {
+        where: {access: 'public'}
+      };
+      db.Document
+        .findAll(query)
+        .then(documents => res.status(200).send(documents));
+    }
+  },
+
+  /**
+   * Retrieve a specific document based on the id
+   * @param {Object} req - Request object
+   * @param {Object} res - Response object
+   * @returns {Object} Response object
+   */
+  retrieve(req, res) {
+    db.Document
+      .findById(req.params.id)
+      .then((document) => {
+        if (!document) {
+          return res.status(404).send({
+            message: 'Document Not Found',
+          });
+        }
+
+        if (Helpers.isAdmin(req)) {
+          return res.status(200).send(document);
+        }
+          
+        if (document.access === 'private' && !Helpers.isCurrentUser(req, document.ownerId)) {
+            return res.status(403)
+              .send({ message: 'This is a private document' });
+        }
+
+        res.send(document);
+      })
+      .catch(error => res.status(400).send(error));
+  },
+
+  /**
+   * Update a document based on the id
+   * @param {Object} req - Request object
+   * @param {Object} res - Response object
+   * @returns {Object} Response object
+   */
+  update(req, res) {
+    db.Document
+      .findById(req.params.id)
+      .then((document) => {
+        if (!document) {
+          return res.status(404).send({
+            message: 'Document Not Found',
+          });
+        }
+
+        if (!Helpers.isAdmin(req) && !Helpers.isCurrentUser(req, document.ownerId)) {
+           return res.status(403)
+            .send({ message: 'This document does not belong to you' });
+        }
+        
+        document
+          .update(req.body, { fields: Object.keys(req.body) })
+          .then(updatedDocument => res.status(200).send(updatedDocument));
+      })
+      .catch(error => res.status(400).send(error));
+  },
+
+  /**
+   * Delete a particular Document
+   * @param {Object} req - Request object
+   * @param {Object} res - Response object
+   * @returns {Object} Response object
+   */
+  delete(req, res) {
+    db.Document
+      .findById(req.params.id)
+      .then((document) => {
+        if (!document) {
+          return res.status(404).send({
+            message: 'This document does not exist',
+          });
+        }
+
+        if (!Helpers.isAdmin(req) && !Helpers.isCurrentUser(req, document.ownerId)) {
+          return res.status(403)
+            .send({ message: 'This document does not belong to you' });
```

Comment:
You don't really need to change this but I think a more generic error message will be better here. E.g. `You are not authorized to delete this document`. Same goes to line  96 and other places like that.